include/deal.II/base/polynomial_space.h

```diff
@@ -141,7 +141,9 @@ class PolynomialSpace
   void compute (const Point<dim>            &unit_point,
                 std::vector<double>         &values,
                 std::vector<Tensor<1,dim> > &grads,
-                std::vector<Tensor<2,dim> > &grad_grads) const;
+                std::vector<Tensor<2,dim> > &grad_grads = Utilities::EmptyVectorGenerator<Tensor<2,dim> >().get_reference(),
+                std::vector<Tensor<3,dim> > &grad_hessians = Utilities::EmptyVectorGenerator<Tensor<3,dim> >().get_reference(),
+                std::vector<Tensor<4,dim> > &grad_hessian_grads = Utilities::EmptyVectorGenerator<Tensor<4,dim> >().get_reference()) const;
```

Comment:
I think that's not great either. You'd also have to make them non-const and mutable (and maybe also static).

My preference would simply be to break compatibility.